### ISSUE TITLE
Performance/Detect Cop Doesn't Catch Violation

### ISSUE THREAD

**Issue by kyledecot** [rubocop-hq/rubocop#6835]
In the following example I would expect that running Rubocop would catch `[].select { true }.first` as it should be using `detect` however it does not.

### Example

```
#!/usr/bin ruby

require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'

  gem 'rubocop'
end

%w[Kyle Decot].select { true }.first

RuboCop::CLI.new.run
```

### Output

```
Fetching gem metadata from https://rubygems.org/...............
Resolving dependencies...
Using ast 2.4.0
Using bundler 2.0.1
Using jaro_winkler 1.5.2
Using parallel 1.14.0
Using parser 2.6.0.0
Using powerpack 0.1.2
Using psych 3.1.0
Using rainbow 3.0.0
Using ruby-progressbar 1.10.0
Using unicode-display_width 1.4.1
Using rubocop 0.65.0
Inspecting 1 file
.

1 file inspected, no offenses detected
```

**Comment by Drenmi** [comment]
Hi, @kyledecot!

This cop has `SafeMode` enabled by default, to prevent some false positives. To disable it, you can configure:

```yml
Performance/Detect:
  Enabled: true
  SafeMode: false
```

After disabling safe mode, the cop correctly registers an offense given your example.
